src/libraries/Microsoft.Extensions.Primitives/src/ChangeToken.cs

```diff
@@ -18,8 +18,17 @@ public static class ChangeToken
         /// <param name="changeTokenProducer">Produces the change token.</param>
         /// <param name="changeTokenConsumer">Action called when the token changes.</param>
         /// <returns></returns>
-        public static IDisposable OnChange(Func<IChangeToken?> changeTokenProducer!!, Action changeTokenConsumer!!)
+        public static IDisposable OnChange(Func<IChangeToken?> changeTokenProducer, Action changeTokenConsumer)
         {
+            if (changeTokenProducer is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.changeTokenProducer);
+            }
+            if (changeTokenConsumer is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.changeTokenConsumer);
+            }
+
```

Comment:
The pattern Microsoft.Extensions.Primitives is using here is slightly more efficient and is what a few libraries like corelib use.  It's using an enum value rather than a string to signal the name of the offending parameter, reducing the size of the call site even further.  We can't remove this pattern without measurements; it's possible it's overkill, it's possible it's meaningful.

Reply:
Oh, that's enlightening. Thanks.